WMIAdventure/frontend/src/js/components/battle/organisms/BattleView/BattleView.js

```diff
@@ -116,42 +134,66 @@ class BattleView extends React.Component {
                                 <ColumnGapContainer gap={'0'}>
                                     <EnemyStateContainer setTranslateX={this.state.enemyStateContainerTranslateX}
                                                          hp={this.state.enemyHp} shield={this.state.enemyShield} />
-                                    <FlexGapContainer setWidth={'100%'} space>
-                                        {/* first card is not visible because is the same as Compact card */}
+                                    <FlexGapContainer setWidth={'100%'} gap={'4px'} reverse>
+                                        {/* Enemy MiniCards!
+                                        First card is not visible because is the same as Compact card */}
                                         {[...Array(5)].map(
                                             (e,i) => {
                                                 return (
                                                     <MiniCardView key={`enemyCard-${i}`}
+                                                                  changeCardsOrder={this.changeCardsEnemyOrder}
+                                                                  cardsOrder={this.state.cardsEnemyOrder[i]}
                                                                   setTranslateX={this.state.enemyMiniCardsTranslateX}
                                                                   enemy cardLevel={this.state.cardLevels[i]}
-                                                                  visible={!(i===0)} animationDuration={`0.${10 - i}`}
-                                                                  cardImage={this.state.icons[5 - i]}/>
+                                                                  animationDuration={`0.${9 - i}`}
+                                                                  cardImage={this.state.icons[i]}/>
                                                 );
                                             }

```

Comment:
Czytelniej by było jakby wyglądało to tak:

```javascript
(e,i) => {
  if(this.state.cardsEnemyOrder[i] !== 1)
    return (
        <MiniCardView key= ... // i tak dalej
    );
}
```

Żeby ogarnąć, jak to się robi to trzeba się przekopać do `MiniCardView`, potem znaleźć, że przekazywany jest ten `cardsOrder` do `CardsContainer` i dopiero tam ogarnąć się, że jest `display: ${({cardsOrder}) => (cardsOrder === 1) ? 'none' : 'flex'}`

A jak się przegląda tylko `CardsContainer`, to się trzeba zastanawiać, po co jest to  `display: ${({cardsOrder}) => (cardsOrder === 1) ? 'none' : 'flex'}`, czemu nie od razu `flex`

No ogólnie komplikacje, keep it simple

Reply:
W tym przypadku, też trzeba przebudować całą mechanike animacji fade, aby działała z tym ifem...

Reply:
W sumie to nie wiem jak to zrobić, aby nie znikał ten komponent instantowo.
Jak kolejność się zmieni, to element nie pasujący do tego ifa zniknie z DOM i nie ma możliwości wtedy animacji opacity.

Reply:
rozumiem, to zostawmy to

Reply:
faktycznie nie pomyślałem o animacjach, mój błąd